model/jpa/src/main/java/org/keycloak/models/jpa/session/JpaUserSessionPersisterProvider.java

```diff
@@ -257,21 +260,32 @@ private static int calculateOldestSessionTime(RealmModel realm, boolean offline)
     private void expire(RealmModel realm, int expired, boolean offline) {
         String offlineStr = offlineToString(offline);
 
-        logger.tracef("Trigger removing expired user sessions for realm '%s'", realm.getName());
+        logger.fatalf("Trigger removing expired user sessions for realm '%s'", realm.getName());
 
-        int cs = em.createNamedQuery("deleteExpiredClientSessions")
+        TypedQuery<Object[]> query = em.createNamedQuery("findExpiredUserSessions", Object[].class)
                 .setParameter("realmId", realm.getId())
                 .setParameter("lastSessionRefresh", expired)
+                .setParameter("offline", offlineStr);
```

Comment:
As a later optimization, you might set `.setHint(AvailableHints.HINT_READ_ONLY, true);` to avoid creating a big persistence context.